dpe/src/x509.rs

```diff
@@ -1088,6 +1299,251 @@ impl X509CertWriter<'_> {
         Ok(bytes_written)
     }
 
+    fn encode_implicit_extensions(
```

Comment:
I wonder if we can avoid duplicating the code with a parameter here. Something like

```rust
enum ConstructedTag {
    Implicit(u8),
    Explicit(u8),
}

fn encode_implicit_extensions(
    &mut self,
    measurements: &MeasurementData,
    constructed: Option<ConstructedTag>,
) -> Result<usize, DpeErrorCode> {
    match constructed {
        Some(Implicit(tag_num)) => // Do implicit tag
        Some(Explicit(tag_num)) => // Do explicit tag
        None => // Use normal tag
    }
}
```

Reply:
Done, except I just passed a bool to encode_extensions since there's only two cases.